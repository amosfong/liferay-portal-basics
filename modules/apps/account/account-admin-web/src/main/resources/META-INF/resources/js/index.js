/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

export {default as AccountContactAddressZipValidator} from './account_entries_admin/AccountContactAddressZipValidator';
export {default as AccountEntriesAdminPortlet} from './account_entries_admin/AccountEntriesAdminPortlet.es';
export {default as AccountEntriesManagementToolbarPropsTransformer} from './account_entries_admin/AccountEntriesManagementToolbarPropsTransformer';
export {default as AccountEntryAddressesManagementToolbarPropsTransformer} from './account_entries_admin/AccountEntryAddressesManagementToolbarPropsTransformer';
export {default as AccountOrganizationsManagementToolbarPropsTransformer} from './account_entries_admin/AccountOrganizationsManagementToolbarPropsTransformer';
export {default as AccountRolesManagementToolbarPropsTransformer} from './account_entries_admin/AccountRolesManagementToolbarPropsTransformer';
export {default as AccountUserDropdownDefaultPropsTransformer} from './account_entries_admin/AccountUserDropdownDefaultPropsTransformer';
export {default as AccountUsersManagementToolbarPropsTransformer} from './account_entries_admin/AccountUsersManagementToolbarPropsTransformer';
export {default as CountryRegionDynamicSelect} from './account_entries_admin/CountryRegionDynamicSelect';
export {default as DefaultAddresses} from './account_entries_admin/DefaultAddresses';
export {default as InviteUsersForm} from './account_entries_admin/InviteUsersForm';
export {default as PersonAccountEntryEventHandler} from './account_entries_admin/PersonAccountEntryEventHandler.es';
export {default as SelectAccountDefaultAddressManagementToolbarPropsTransformer} from './account_entries_admin/SelectAccountDefaultAddressManagementToolbarPropsTransformer';
export {default as SelectAccountUsersManagementToolbarPropsTransformer} from './account_entries_admin/SelectAccountUsersManagementToolbarPropsTransformer';
export {default as AccountGroupAccountEntriesManagementToolbarPropsTransformer} from './account_groups_admin/AccountGroupAccountEntriesManagementToolbarPropsTransformer';
export {default as AccountGroupsManagementToolbarPropsTransformer} from './account_groups_admin/AccountGroupsManagementToolbarPropsTransformer';
export {default as AccountUserEmailDomainValidator} from './account_users_admin/AccountUserEmailDomainValidator.es';
export {default as AccountUsersAdminManagementToolbarPropsTransformer} from './account_users_admin/AccountUsersAdminManagementToolbarPropsTransformer';
