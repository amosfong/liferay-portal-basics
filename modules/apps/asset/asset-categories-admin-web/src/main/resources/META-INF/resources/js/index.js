/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

export {default as ActionsComponentPropsTransformer} from './ActionsComponentPropsTransformer';
export {default as AssetCategoriesManagementToolbarPropsTransformer} from './AssetCategoriesManagementToolbarPropsTransformer';
export {default as AssetCategoriesSelectorTag} from './AssetCategoriesSelectorTag';
export {default as CategoryActionDropdownPropsTransformer} from './CategoryActionDropdownPropsTransformer';
export {default as ItemSelectorAddCategory} from './ItemSelectorAddCategory';
export {default as SaveAndAddNewPropsTransformer} from './SaveAndAddNewPropsTransformer';
export {default as VocabularyActionDropdownPropsTransformer} from './VocabularyActionDropdownPropsTransformer';
export {default as openDeleteCategoryModal} from './openDeleteCategoryModal';
export {default as openDeleteVocabularyModal} from './openDeleteVocabularyModal';
