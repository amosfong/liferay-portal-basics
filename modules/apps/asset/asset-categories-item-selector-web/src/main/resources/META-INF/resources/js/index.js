/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

export {AssetCategoryTree} from './AssetCategoryTree';
export {default as SelectAssetCategoryInfoItem} from './SelectAssetCategoryInfoItem';
export {default as SelectEntityHandler} from './SelectEntityHandler';
