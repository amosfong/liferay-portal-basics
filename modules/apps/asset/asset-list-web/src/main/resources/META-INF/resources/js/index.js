/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

export {default as AssetEntryListDropdownDefaultPropsTransformer} from './AssetEntryListDropdownDefaultPropsTransformer';
export {default as AssetListEntryVariationDefaultPropsTransformer} from './AssetListEntryVariationDefaultPropsTransformer';
export {default as EditAssetListEntryManualDefaultPropsTransformer} from './EditAssetListEntryManualDefaultPropsTransformer';
export {default as EmptyResultMessagePropsTransformer} from './EmptyResultMessagePropsTransformer';
export {default as InfoCollectionProviderDropdownDefaultPropsTransformer} from './InfoCollectionProviderDropdownDefaultPropsTransformer';
export {default as ListItemsDropdownPropsTransformer} from './ListItemsDropdownPropsTransformer';
export {default as ManagementToolbarPropsTransformer} from './ManagementToolbarPropsTransformer';
export {default as Ordering} from './Ordering';
export {default as ScopeDefaultPropsTransformer} from './ScopeDefaultPropsTransformer';
export {default as SelectStructureField} from './SelectStructureField';
export {default as Source} from './Source';
export {default as TopLinkEventHandler} from './TopLinkEventHandler';
export {default as AssetFilterBuilder} from './components/AssetFilterBuilder/index';
export {default as VariationsNav} from './components/VariationsNav/index';
export {default as openDeleteAssetEntryListModal} from './openDeleteAssetEntryListModal';
