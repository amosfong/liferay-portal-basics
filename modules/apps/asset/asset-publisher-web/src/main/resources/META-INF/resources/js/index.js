/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

export {default as AssetEntryActionButtonPropsTransformer} from './AssetEntryActionButtonPropsTransformer';
export {default as AssetEntryActionDropdownPropsTransformer} from './AssetEntryActionDropdownPropsTransformer';
export {default as AssetEntrySelectionDropdownPropsTransformer} from './AssetEntrySelectionDropdownPropsTransformer';
export {default as CreateAssetListActionButtonPropsTransformer} from './CreateAssetListActionButtonPropsTransformer';
export {default as DisplaySettings} from './DisplaySettings';
export {default as Ordering} from './Ordering';
export {default as ScopeActionDropdownPropsTransformer} from './ScopeActionDropdownPropsTransformer';
export {default as SelectCollection} from './SelectCollection';
export {default as SelectStructureField} from './SelectStructureField';
export {default as Source} from './Source';
export {default as AssetFilterBuilder} from './components/AssetFilterBuilder/index';
export {default as printPageButtonPropsTransformer} from './printPageButtonPropsTransformer';
