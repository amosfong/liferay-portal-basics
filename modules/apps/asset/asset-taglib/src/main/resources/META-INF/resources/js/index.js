/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

export {default as InputAssetLinkDropdownDefaultPropsTransformer} from './InputAssetLinkDropdownDefaultPropsTransformer';
export {default as AssetCategoriesNavigationTreeView} from './asset_categories_navigation/AssetCategoriesNavigationTreeView';
export {default as AssetCategoriesSelector} from './asset_categories_selector/AssetCategoriesSelector';
export {default as AssetCategoriesSelectorTag} from './asset_categories_selector/AssetCategoriesSelectorTag';
export {default as AssetVocabularyCategoriesSelector} from './asset_categories_selector/AssetVocabularyCategoriesSelector';
export {ASSET_VOCABULARY_VISIBILITY_TYPES} from './asset_categories_selector/assetVocabularyVisibilityTypes';
export {default as AssetTagsSelector} from './asset_tags_selector/AssetTagsSelector';
export {default as AssetTagsSelectorTag} from './asset_tags_selector/AssetTagsSelectorTag';
