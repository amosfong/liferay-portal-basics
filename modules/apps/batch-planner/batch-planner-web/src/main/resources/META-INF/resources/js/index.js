/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

export {default as BatchPlannerPlanTemplateManagementToolbarPropsTransformer} from './BatchPlannerPlanTemplateManagementToolbarPropsTransformer';
export {default as DownloadHelper} from './DownloadHelper';
export {default as FieldsTable} from './FieldsTable';
export {default as SaveTemplate} from './SaveTemplate';
export {default as ExportSettings} from './components/ExportSettings';
export {default as FileUpload} from './components/FileUpload';
export {default as ImportEntityType} from './components/ImportEntityType';
export {default as Scope} from './components/Scope';
export {default as Strategies} from './components/Strategies';
export {default as editBatchPlannerPlan} from './editBatchPlannerPlan';
export {default as Export} from './export/Export';
export {default as ImportForm} from './import/ImportForm';
export {default as showUploadInput} from './showUploadInput';
