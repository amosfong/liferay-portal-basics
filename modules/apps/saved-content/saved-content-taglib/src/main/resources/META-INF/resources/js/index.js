/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

import {ClayButtonWithIcon} from '@clayui/button';
import {fetch, objectToFormData, openToast, sub} from 'frontend-js-web';
import PropTypes from 'prop-types';
import React, {useState} from 'react';

function showNotification(message, error = false) {
	openToast({
		message,
		type: error ? 'danger' : undefined,
	});
}

export function SavedContentEntry({
	className,
	classPK,
	contentTitle,
	enabled = false,
	mySavedContentURL,
	portletNamespace,
	saved: initialSaved = false,
	savedContentEntryURL,
}) {
	const [saved, setSaved] = useState(initialSaved);
	const [loading, setLoading] = useState(false);

	const handleSubmit = (event) => {
		event.preventDefault();

		setLoading(true);
		setSaved((saved) => !saved);

		fetch(savedContentEntryURL, {
			body: objectToFormData({
				[`${portletNamespace}className`]: className,
				[`${portletNamespace}classPK`]: classPK,
			}),
			method: 'POST',
		})
			.then((response) => response.json())
			.then((response) => {
				if (response.errorMessage) {
					setSaved((saved) => !saved);
					showNotification(response.errorMessage, true);

					return;
				}

				setSaved(response.saved);

				const mySavedContentLink = `
					<a href="${mySavedContentURL}" class="alert-link">${Liferay.Language.get(
						'my-saved-content'
					)}</a>
				`.trim();

				showNotification(
					sub(
						saved
							? Liferay.Language.get(
									'x-has-been-successfully-removed-from-x'
								)
							: Liferay.Language.get('x-has-been-saved-in-x'),
						contentTitle,
						mySavedContentLink
					)
				);
			})
			.catch(() => {
				setSaved((saved) => !saved);
				showNotification(
					Liferay.Language.get('an-unexpected-error-occurred'),
					true
				);
			})
			.finally(() => {
				setLoading(false);
			});
	};

	return (
		<form onSubmit={handleSubmit}>
			<ClayButtonWithIcon
				aria-label={
					saved
						? sub(Liferay.Language.get('remove-x'), contentTitle)
						: sub(Liferay.Language.get('save-x'), contentTitle)
				}
				disabled={!enabled || loading}
				displayType="secondary"
				monospaced
				size="sm"
				symbol={saved ? 'bookmarks-full' : 'bookmarks'}
				title={
					saved
						? Liferay.Language.get('remove-content')
						: Liferay.Language.get('save-content')
				}
				type="submit"
			/>
		</form>
	);
}

SavedContentEntry.propTypes = {
	className: PropTypes.string.isRequired,
	classPK: PropTypes.string.isRequired,
	contentTitle: PropTypes.string.isRequired,
	enabled: PropTypes.bool,
	mySavedContentURL: PropTypes.string.isRequired,
	portletNamespace: PropTypes.string.isRequired,
	saved: PropTypes.bool,
	savedContentEntryURL: PropTypes.string.isRequired,
};
