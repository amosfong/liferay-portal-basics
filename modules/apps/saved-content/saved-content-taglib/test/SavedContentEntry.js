/**
 * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
 * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
 */

import {act, render, screen} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';

import '@testing-library/jest-dom/extend-expect';
import {
	fetch as mockedFetch,
	openToast as mockedOpenToast,
} from 'frontend-js-web';

import {SavedContentEntry} from '../src/main/resources/META-INF/resources/js/index';

jest.mock('frontend-js-web', () => ({
	...jest.requireActual('frontend-js-web'),
	fetch: jest.fn(),
	openToast: jest.fn(),
	sub: jest.fn((langKey, arg) => langKey.replace('x', arg)),
}));

function renderComponent(props) {
	return render(
		<SavedContentEntry
			className="className"
			classPK="classPK"
			contentTitle="contentTitle"
			enabled
			mySavedContentURL="/my-saved-content-url"
			portletNamespace="namespace"
			savedContentEntryURL="/saved-content-entry-url"
			{...props}
		/>
	);
}

describe('SavedContentEntry', () => {
	afterEach(() => {
		jest.clearAllMocks();
	});

	describe('render', () => {
		it('disabled', () => {
			const {container} = renderComponent({enabled: false});
			const button = screen.getByRole('button');

			expect(container).toMatchSnapshot();
			expect(button).toBeDisabled();
		});

		it('enabled', () => {
			const {container} = renderComponent();
			const button = screen.getByRole('button');

			expect(container).toMatchSnapshot();
			expect(button).toBeEnabled();
		});
	});

	it('calls backend with the given config', async () => {
		mockedFetch.mockReturnValue(
			Promise.resolve({
				json: () => Promise.resolve({saved: true}),
			})
		);

		renderComponent();

		await act(async () => {
			userEvent.click(screen.getByRole('button'));
		});

		expect(mockedFetch).toHaveBeenCalledTimes(1);

		const [[url, {body}]] = mockedFetch.mock.calls;

		expect(url).toBe('/saved-content-entry-url');
		expect(body.get('namespaceclassName')).toBe('className');
		expect(body.get('namespaceclassPK')).toBe('classPK');
	});

	it('saves the content and notify', async () => {
		mockedFetch.mockReturnValue(
			Promise.resolve({
				json: () => Promise.resolve({saved: true}),
			})
		);

		renderComponent();

		const button = screen.getByRole('button', {
			name: 'save-contentTitle',
		});

		await act(async () => {
			userEvent.click(button);
		});

		expect(button).toHaveAttribute('aria-label', 'remove-contentTitle');
		expect(mockedOpenToast.mock.calls[0][0].message).toBe(
			'contentTitle-has-been-saved-in-x'
		);
	});

	it('removes the content and notify', async () => {
		mockedFetch.mockReturnValue(
			Promise.resolve({
				json: () => Promise.resolve({saved: false}),
			})
		);

		renderComponent({saved: true});

		const button = screen.getByRole('button', {
			name: 'remove-contentTitle',
		});

		await act(async () => {
			userEvent.click(button);
		});

		expect(button).toHaveAttribute('aria-label', 'save-contentTitle');
		expect(mockedOpenToast.mock.calls[0][0].message).toBe(
			'contentTitle-has-been-successfully-removed-from-x'
		);
	});
});
