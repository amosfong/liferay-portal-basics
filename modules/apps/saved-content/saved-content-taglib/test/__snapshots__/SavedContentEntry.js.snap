// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedContentEntry render disabled 1`] = `
<div>
  <form>
    <button
      aria-label="save-contentTitle"
      class="btn btn-monospaced btn-sm btn-secondary"
      disabled=""
      title="save-content"
      type="submit"
    >
      <svg
        class="lexicon-icon lexicon-icon-bookmarks"
        role="presentation"
      >
        <use
          href="#bookmarks"
        />
      </svg>
    </button>
  </form>
</div>
`;

exports[`SavedContentEntry render enabled 1`] = `
<div>
  <form>
    <button
      aria-label="save-contentTitle"
      class="btn btn-monospaced btn-sm btn-secondary"
      title="save-content"
      type="submit"
    >
      <svg
        class="lexicon-icon lexicon-icon-bookmarks"
        role="presentation"
      >
        <use
          href="#bookmarks"
        />
      </svg>
    </button>
  </form>
</div>
`;
